### DISCUSSION TITLE
snap is not working inside github codespaces

### DISCUSSION THREAD

**Discussion by ratbekn** [community/community#54655] [Codespaces] [answered]
### Select Topic Area

Question

### Body

What I want:
Install snap (snapcraft.io)

What I did:
1. Create new codespace using github cli - `gh codespace create`
2. Connect to codespace using github cli - `gh codespace ssh`
3. Install snap using apt - `sudo apt update && sudo apt install snapd`

What I expect:
Installed working snap

What I got:
1. on executing any snap command:
```
error: cannot *command*  snaps: cannot communicate with server: Get http://localhost/v2/snaps: dial unix /run/snapd.socket: connect: no such file or directory
```
2. `snap version` states: 
```
snap    2.58+20.04
snapd   unavailable
series  -
```
3.  there is no `snapd` in `service --status-all`
5.  `service snapd status` gives `snapd: unrecognized service`

Ubuntu version: `Ubuntu 20.04.6 LTS (GNU/Linux 5.4.0-1106-azure x86_64)`

How to fix that?

**Comment by codergautam** [comment]
any update?

**Comment by shivaylamba** [comment]
any update here?

**Comment by shivaylamba** [comment]
@mgriffin

**Comment by alessandro-newzoo** [comment]
jesus can we get some help? it's been 5 months

**Comment by mgriffin** [comment, marked as answer]
Codespaces instances are [hosted in a Docker container](https://docs.github.com/en/codespaces/overview) and because of this `systemd` isn't used. `systemd` is a requirement for `snapd` to run, which I found out from [this StackOverflow answer](https://stackoverflow.com/a/58484154).

In short, it's doesn't seem like it's possible to run `snap` inside of GitHub Codespace instances.

**Reply by alessandro-newzoo** [reply]
that's a real bummer :( thank you for looking into that.

**Reply by shivaylamba** [reply]
What sort of Linux package manager can we use to be supported in GitHub Codespaces

**Reply by niemeyer** [reply]
As I understand it it's actually a container inside a VM, for proper isolation. It would be great to have access to the actual underlying VM for a more complete environment.

**Comment by liudonghua123** [comment]
How about convert systemd script (`/lib/systemd/system/snapd.service`) to init.d script (`/etc/init.d/snapd`).

```shell
$ cat /lib/systemd/system/snapd.service
[Unit]
Description=Snap Daemon
After=snapd.socket
After=time-set.target
After=snapd.mounts.target
Wants=time-set.target
Wants=snapd.mounts.target
Requires=snapd.socket
OnFailure=snapd.failure.service
# This is handled by snapd
# X-Snapd-Snap: do-not-start

[Service]
# Disabled because it breaks lxd
# (https://bugs.launchpad.net/snapd/+bug/1709536)
#Nice=-5
OOMScoreAdjust=-900
ExecStart=/usr/lib/snapd/snapd
EnvironmentFile=-/etc/environment
Restart=always
WatchdogSec=5m
Type=notify
SuccessExitStatus=42
RestartPreventExitStatus=42
KillMode=process

[Install]
WantedBy=multi-user.target
```

But it has some limits, see also https://forum.snapcraft.io/t/installing-and-running-snaps-without-systemd/3949/36 ...

Update: I tried to install [flutter](https://docs.flutter.dev/get-started/install/linux) in github codespaces, and after I installed [snapd](https://snapcraft.io/docs/installing-snap-on-ubuntu), I failed to start snapd, then found this discussion. Maybe I have to use [method-2-manual-installation](https://docs.flutter.dev/get-started/install/linux#method-2-manual-installation).
